pkg/watcher/internal/metricsprovider/prometheus.go

```diff
@@ -84,37 +96,12 @@ func NewPromClient(opts watcher.MetricsProviderOpts) (watcher.MetricsProviderCli
 	// Check if EnableOpenShiftAuth is set.
 	_, enableOpenShiftAuth := os.LookupEnv(EnableOpenShiftAuth)
 	if enableOpenShiftAuth {
-		// Create the config for kubernetes client
-		clusterConfig, err := rest.InClusterConfig()
-		if err != nil {
-			// Get the kubeconfig path
-			kubeConfigPath, ok := os.LookupEnv(kubeConfig)
-			if !ok {
-				kubeConfigPath = defaultKubeConfig
-			}
-			clusterConfig, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
-			if err != nil {
-				return nil, fmt.Errorf("failed to get kubernetes config: %v", err)
-			}
-		}
-
-		// Create the client for kubernetes
-		kclient, err := kubernetes.NewForConfig(clusterConfig)
+		// Retrieve Pod CA cert
+		caCertPool, err := loadCAFile(K8sPodCAFilePath)
 		if err != nil {
-			return nil, fmt.Errorf("failed to create kubernetes client: %v", err)
+			log.Fatalf("Error loading CA file: %v", err)
```

Comment:
why do we use log.Fatalf but not return error? I think we should make consistent across the metrics clients.

Reply:
Yes, @lenhattan86 , you are right. I am going to change it. Still working on it and will share when I am ready.